test/bigquery.ts

```diff
@@ -1291,6 +1323,53 @@ describe('BigQuery', () => {
             assert.ifError
           );
         });
+
+        it('should allow for optional parameter types', done => {
+          bq.createJob = (reqOpts: JobOptions) => {
+            // tslint:disable-next-line no-any
+            assert.strictEqual((reqOpts as any).params, undefined);
+            done();
+          };
+
+          bq.createQueryJob(
+            {
+              query: QUERY_STRING,
+              params: NAMED_PARAMS,
+              types: NAMED_TYPES,
+            },
+            assert.ifError
+          );
+        });
+
+        it('should throw for invalid provided types', done => {
+          bq.createJob = (reqOpts: JobOptions) => {
+            done();
+          };
+
+          const INVALID_TYPES = {key: 'invalid'};
+
+          assert.throws(() => {
+            bq.createQueryJob({
+              query: QUERY_STRING,
+              params: NAMED_PARAMS,
+              types: INVALID_TYPES,
+            });
+          }, /Invalid type provided./);
+        });
+
+        it('should throw for invalid type structure provided', done => {
+          bq.createJob = (reqOpts: JobOptions) => {
```

Comment:
I would remove this override since this hopefully won't get called. Another option would be executing done with an error, `done(new Error('Should not have been called.'))`.